src/Bicep.Core/Registry/ArtifactType.cs

```diff
@@ -7,5 +7,6 @@ public enum ArtifactType
     {
         Module,
         Provider,
+        Type
```

Comment:
I don't think you need to introduce a new concept here - `Provider` was introduced already for this purpose, so you should use that instead. We have been referring to these as "provider type packages".

Generally I'd recommend renaming terminology from "type" -> "provider" to make this clearer - e.g.:
`bicep publish-provider`
`PublishProviderCommand` etc.